proxy.go

```diff
@@ -132,13 +133,56 @@ func setupLogger(logLevel string) error {
 	return nil
 }
 
+func printAgentLogs(sock string) error {
+	// Don't return an error if nothing has been provided.
+	// This flag is optional.
+	if sock == "" {
+		return nil
+	}
+
+	agentLogsAddr, err := unixAddr(sock)
+	if err != nil {
+		logger().WithField("socket-address", sock).Fatal("invalid agent logs socket address")
+		return err
+	}
+
+	// Check permissions socket for "other" is 0.
+	fileInfo, err := os.Stat(agentLogsAddr)
+	if err != nil {
+		return err
+	}
+	remainder := fileInfo.Mode().Perm() % 10
```

Comment:
I'd rather we ensured that "other" had no bits set at all. Something like:

```
perms := st.Mode() & os.ModePerm 
otherMask := 0007

other := int(perms) & otherMask

if other != 0 {
   return fmt.Errorf("All socket permissions for 'other' should be disabled, got %3.3o", other)
}


```

Reply:
@jodh-intel that's what I am doing using the division by 10 here. I make sure the other bit is 0, otherwise I generate an error.
Are you sure what you're proposing is working here ? I mean we are not manipulating real bits...